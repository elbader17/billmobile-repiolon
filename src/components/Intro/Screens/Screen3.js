import React, { Component } from 'react';
import { Text, View, Alert } from 'react-native';
import Button from 'react-native-button'; //https://www.npmjs.com/package/react-native-button
import style from './style';

class Screen3 extends Component {
  _onPressButton() {
    Alert.alert('You tapped the button!')
  }
   
  render() {
    return (
      <View style={style.slide3}>
        <Text style={style.number}>3</Text>
        <Text style={style.textTittle}>Información segura</Text>
        <Text style={style.textDescription}>Nuestra política de privacidad es simple: No 
        compartimos ninguna información personal o impositiva tuya.</Text>
        <Button 
          onPress={this._onPressButton} 
          style={style.styleButton} 
          containerStyle={style.positionButtonEnd}
          testID={'ready'}
        >
        Listo
        </Button>
      </View>
    );
  }
}

export default Screen3;
  