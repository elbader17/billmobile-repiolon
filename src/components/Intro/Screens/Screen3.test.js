import React from 'react';
import { Alert } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Screen3 from './Screen3';

describe('Rendering Component Screen3', () => {
  test('renders correctly', () => {
    const component = renderer.create(<Screen3 />).toJSON();
    expect(component).toMatchSnapshot();
  });
})

describe('Screen3 Button', () => {
  it('Screen3', () => {
    const clickFn = jest.fn();
    const { alert } = Alert;
    Alert.alert = clickFn;
    const wrapper = shallow(<Screen3 />);
    const button = wrapper.findWhere(node => node.prop('testID') === 'ready');
    button.simulate('press');
    Alert.alert = alert;
    expect(clickFn).toHaveBeenCalled();
  })
});
