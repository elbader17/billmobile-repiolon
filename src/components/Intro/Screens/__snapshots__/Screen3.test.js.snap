// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Component Screen3 renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3687d1",
      "height": 1334,
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <Text
    style={
      Object {
        "bottom": 422,
        "color": "#fff",
        "fontFamily": undefined,
        "fontSize": 80,
        "lineHeight": 90,
        "position": "absolute",
      }
    }
  >
    3
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#fff",
        "fontFamily": undefined,
        "fontSize": 18,
        "height": 33,
        "lineHeight": 32,
        "position": "absolute",
        "textAlign": "center",
        "top": 287,
        "width": 289,
      }
    }
  >
    Información segura
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "color": "#fff",
        "fontFamily": undefined,
        "fontSize": 14,
        "height": 66,
        "lineHeight": 21,
        "position": "absolute",
        "textAlign": "center",
        "top": 332,
        "width": 289,
      }
    }
  >
    Nuestra política de privacidad es simple: No compartimos ninguna información personal o impositiva tuya.
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "opacity": 1,
        "top": 190,
      }
    }
    testID="ready"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007aff",
            "fontSize": 17,
            "fontWeight": "500",
            "textAlign": "center",
          },
          null,
          Object {
            "borderColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "color": "#fff",
            "fontFamily": undefined,
            "fontSize": 14,
            "height": 40,
            "padding": 10,
            "textAlign": "center",
            "width": 247,
          },
          null,
        ]
      }
    >
      Listo
    </Text>
  </View>
</View>
`;
